lib/node/runner/checker_ballot_transaction.go

```diff
@@ -310,16 +308,16 @@ func ValidateOp(st *storage.LevelDBBackend, source *block.BlockAccount, op opera
 		if taccount.Linked != "" {
 			return errors.FrozenAccountNoDeposit
 		}
+
+		// The source account is frozen account
 		if source.Linked != "" {
 			// If it's a frozen account, everything must be withdrawn
-			var expected common.Amount
-			expected, err = source.Balance.Sub(common.BaseFee)
-			if casted.Amount != expected {
+			if source.Balance != casted.Amount { // No fee for Unfreeze Payment
 				return errors.FrozenAccountMustWithdrawEverything
 			}
 			// Unfreezing must be done after X period from unfreezing request
 			iterFunc, closeFunc := block.GetBlockOperationsBySource(st, source.Address, nil)
-			bo, _, _ := iterFunc()
+			bo, _, _ := iterFunc() //Get the first operation submitted by the source(frozen) account
```

Comment:
It is possible that `bo is nil`?

Reply:
https://github.com/bosnet/sebak/blob/13dbe0a17626858b51b61b4dde078554b5597458/lib/block/operation.go#L229-233
It always returns not nil.
If there is no item then BlockOperation{} is returned